service-worker.js

```diff
@@ -90,6 +91,85 @@ self.addEventListener('activate', event => {
   }());
 });
 
+/**
+ * Helper to know if the request is an image request
+ * @param {Object} request The event request object
+ * @returns {boolean} Is the request an image request?
+ */
+const isImageRequest = request => IMAGE_REGEX.test(request.url);
+
+/**
+ * Handle the `fetch` service worker event.
+ */
 self.addEventListener('fetch', event => {
-  console.log('Fetch occurred for:', event.request.url);
+  const request = event.request;
+  const requestURL = new URL(request.url);
+
+  // Local URLs
+  if (requestURL.origin === location.origin) {
+    /**
+     * Images
+     * Stale-While-Revalidate
+     */
+    if (isImageRequest(request)) {
+      event.respondWith(async function() {
+        const cachedImage = await caches.match(request);
+        if (cachedImage) {
+          console.info('Cache Fetch:', request.url);
+          return cachedImage;
+        }
+        
+        const networkImage = await fetch(request);
+        const imagesCache = await caches.open(CACHES.IMAGES);
+        event.waitUntil(
+          imagesCache.put(request, networkImage.clone())
+        );
+
+        console.info('Network Fetch:', request.url);
+        return networkImage;
```

Comment:
@grigs FYI there will be a follow-up PR to refactor this a bit to more closely match Jake's example:

```js
self.addEventListener('fetch', (event) => {
  event.respondWith(async function() {
    const cache = await caches.open('mysite-dynamic');
    const cachedResponse = await cache.match(event.request);
    const networkResponsePromise = fetch(event.request);

    event.waitUntil(async function() {
      const networkResponse = await networkResponsePromise;
      await cache.put(event.request, networkResponse.clone());
    }());

    // Returned the cached response if we have one, otherwise return the network response.
    return cachedResponse || networkResponsePromise;
  }());
});
```
I like how he keeps the `return cachedResponse || networkResponsePromise` together.

https://jakearchibald.com/2014/offline-cookbook/#stale-while-revalidate

Reply:
Also, [see my other comment](https://github.com/cloudfour/pwa-workshop/pull/9#issuecomment-496559081).